ext-dynamic-commitments.md

```diff
@@ -0,0 +1,1087 @@
+and Schnorr contexts.
+
+#### Extensions to `dyn_ack`:
+
+1. `tlv_stream`: `dyn_ack_tlvs`
+2. types:
+    1. type: 0 (`local_musig2_pubnonce`)
+    2. data:
+        * [`66*byte`:`nonces`]
+
+#### Requirements
+
+The sending node:
+  - if it is accepting a `channel_type` of `simple_taproot_channel`:
+    - MUST set `local_musig2_pubnonce` to the nonce that it will use to verify
+      local commitments.
+
+The receiving node:
+  - MUST send an `error` and fail the channel if `local_musig2_pubnonce` cannot
+    be parsed as two compressed secp256k1 points.
+
+# Appendix
+
+## Pinning
+
+![Cannot display image](./dynamic-commits/kickoff%20pinning.png "Kickoff transaction pinned")
+
+Originally, Bitcoin Core's default mempool settings allowed an unconfirmed
+transaction to have up to 25 decendants in the mempool. Past this limit, any
+descendants would be rejected. This was used as a DoS mitigation in Bitcoin Core
+and affected the security of LN channels. Before the anchors commitment type was
+introduced, pinning in the LN was where a counterparty broadcasted the
+commitment transaction and created a chain of 25 descendants spending from one
+of the commitment's outputs. The time-sensitive commitment transaction could be
+"pinned" to the bottom of the mempool.  This was addressed with a change to
+Bitcoin Core called CPFP Carve-out.
+
+### CPFP Carve-out
+
+CPFP Carve-out was introduced to Bitcoin Core in
+https://github.com/bitcoin/bitcoin/pull/15681. If a Bitcoin node receives a
+transaction that is rejected due to any of the mempool size or
+ancestor/descendant restrictions being hit, it will try to accept the
+transaction again. This second try will succeed only if:
+  - the transaction is 40kWU or less
+  - it has only one ancestor in the mempool
+
+This change, in conjunction with the anchor commitment type, decreases the
+efficacy of the pinning attack since the honest party can still attach an anchor
+despite the descendant size limit being hit.
+
+### Dynamic Commitments & CPFP Carve-out
+
+The safety guarantees of CPFP Carve-out break due to the structure of the
+kickoff transaction.  The kickoff transaction contains 3 spendable outputs: the
+local party's anchor, the remote party's anchor, and the new funding output. All
+three of these outputs can be spent immediately.  A malicious counterparty can
+pin the kickoff transaction by:
+  - spending from their anchor output to create a descendant chain of 25
+    transactions _AND_
+  - spending from the new funding output using the new commitment transaction,
+    "using up" the CPFP Carve-out slot designated for the honest party.
+_NOTE FOR REVIEWERS_: The semantics of CPFP carve-out are not entirely clear
+as to whether or not there is only _one_ CPFP-Carve-Out "slot" or if the only
+two requirements are the 40kWU limit and a single unconfirmed ancestor. If we
+have more than one "slot" available, this is no longer a concern.
```

Comment:
The carve out is "just one more", so it only works if each party has only one spendable output otherwise your counterparty can occupy the carve out slot themselves. Was recently looking at this myself and dug up [this conversation](https://github.com/lightning/bolts/pull/688#issuecomment-549480380) from the original anchors PR.